kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java

```diff
@@ -289,8 +289,8 @@ public void execute() {
                 // An unexpected error occurred and we didn't even get an onFailure callback.
                 logger.error("Exception in reconnect", e);
                 webSocketRef.set(null);
+                closeEvent(new KubernetesClientException("Unhandled exception in reconnect attempt", e));
```

Comment:
is the specific fix this reordering before the `close()`?  the other changes look like a refactoring / log level adjustments